### PR TITLE
Adding image substitution logic to bake action

### PR DESCRIPTION
**Pull request by shigupt202**
Fixes https://github.com/microsoft/azure-pipelines-tasks/issues/12610

### CODE REVIEW

File: Tasks/KubernetesManifestV0/src/actions/bake.ts [hunk 1]
```diff
@@ -12,12 +13,25 @@ import { Helm, NameValuePair } from 'kubernetes-common-v2/helm-object-model';
...
     }
+    protected updateImages(filePath: string) {
+        if (TaskInputParameters.containers) {
+            const updatedFilesPaths: string[] = DeploymentHelper.updateResourceObjects([filePath], [], TaskInputParameters.containers);
+            let fileContents: string[] = [];
+            updatedFilesPaths.forEach((path) => {
+                const content = yaml.safeDump(JSON.parse(fs.readFileSync(path).toString()));
+                fileContents.push(content);
+            });
+            fs.writeFileSync(filePath, fileContents.join("\n---\n"));
```
Comment (ds-ms) [inline Tasks/KubernetesManifestV0/src/actions/bake.ts:32]: Is there no function in the yaml lib to do the concatenation for you?

File: Tasks/KubernetesManifestV0/task.json [hunk 2]
```diff
@@ -137,10 +137,10 @@
             "type": "pickList",
             "label": "Render Engine",
             "required": false,
-            "defaultValue": "helm",
+            "defaultValue": "helm2",
```
Comment (ds-ms) [inline Tasks/KubernetesManifestV0/task.json:140]: Make this helm everywhere instead.

File: Tasks/KubernetesManifestV0/src/actions/bake.ts (hunk 1 above)
Comment (shigupt202) [reply Tasks/KubernetesManifestV0/src/actions/bake.ts:32]: Couldn't find anything. I tried doing yaml.safeDump on array of JSON objects. But that results in a yaml file without any separator `---`.

File: Tasks/KubernetesManifestV0/task.json (hunk 2 above)
Comment (shigupt202) [reply Tasks/KubernetesManifestV0/task.json]: Done

File: Tasks/KubernetesManifestV0/Tests/L0.ts [hunk 3]
```diff
@@ -357,6 +358,21 @@ describe('Kubernetes Manifests Suite', function () {
...
 
+    it('Run should succeed with helm bake with image substituion', (done: MochaDone) => {
+        const tp = path.join(__dirname, 'TestSetup.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        process.env[shared.TestEnvVars.action] = shared.Actions.bake;
+        process.env[shared.TestEnvVars.namespace] = 'namespacefrominput';
+        process.env[shared.TestEnvVars.helmChart] = 'helmChart';
+        process.env[shared.TestEnvVars.renderType] = 'helm';
+        process.env[shared.TestEnvVars.helmVersion] = "v2";
+        process.env[shared.TestEnvVars.containers] = 'nginx:1.1.1';
```
Comment (ds-ms) [inline Tasks/KubernetesManifestV0/Tests/L0.ts:369]: Add a test case which covers multiple container replacements as well

File: Tasks/KubernetesManifestV0/src/actions/bake.ts (hunk 1 above)
Comment (ds-ms) [reply Tasks/KubernetesManifestV0/src/actions/bake.ts:32]: Add a test case where you use read the generated file. i.e. if `yaml.safeLoad` works

File: Tasks/KubernetesManifestV0/Tests/L0.ts (hunk 3 above)
Comment (shigupt202) [reply Tasks/KubernetesManifestV0/Tests/L0.ts:369]: Done

**Pull request merged**